src/heads-up/views/linkedin.vue

```diff
@@ -0,0 +1,155 @@
+          Preview based on
+          <external-link href="https://linkedin.com/"
+            >linkedin.com</external-link
+          >.
+        </figcaption>
+      </figure>
+    </panel-section>
+
+    <panel-section title="Properties">
+      <properties-list>
+        <template v-if="og.title">
+          <dt>og:title</dt>
+          <dd>{{ og.title }}</dd>
+        </template>
+        <template v-if="og.image">
+          <dt>og:image</dt>
+          <dd>
+            <external-link :href="absoluteUrl(og.image)">
+              <img alt :src="absoluteUrl(og.image)" />
+              <span>{{ og.image }}</span>
+            </external-link>
+            <p v-if="imageDimensions">
+              ({{ imageDimensions.width }} x {{ imageDimensions.height }}px)
+            </p>
+          </dd>
+        </template>
+      </properties-list>
+    </panel-section>
+  </div>
+</template>
+
+<script>
+import { mapState } from "vuex";
+import { ExternalLink, PanelSection, PropertiesList } from "../components";
+import {
+  findMetaContent,
+  findMetaProperty,
+  findImageDimensions
+} from "../lib/find-meta";
+
+function getHostName(url) {
+  if (!url) {
+    return "";
+  }
+  const hostname = new URL(url).hostname;
+  const wwwPrefix = "www.";
+  return hostname.startsWith(wwwPrefix)
+    ? hostname.slice(wwwPrefix.length)
+    : hostname;
+}
+
+export default {
+  components: { ExternalLink, PanelSection, PropertiesList },
+  data() {
+    return {
+      iframeHeight: "auto",
+      imageDimensions: { width: undefined, height: undefined },
+      previewUrl: ""
+    };
+  },
+  computed: {
+    ...mapState(["head"]),
+    hasOgImage() {
+      return this.og.image ? true : false;
+    },
+
+    og() {
+      return {
+        title: this.propertyValue("og:title") || this.head.title || "Weblink",
+        image: this.absoluteUrl(this.propertyValue("og:image"))
+      };
+    }
+  },
+  mounted() {
+    window.addEventListener("resize", this.onResize);
+  },
+  created() {
+    findImageDimensions(this.head, "og:image").then(imageDimensions => {
+      this.imageDimensions = imageDimensions;
+      this.previewUrl = this.getPreviewUrl({ imageDimensions });
+
+      if (!imageDimensions.correctImgUrl) {
+        console.log(`og.image can't be loaded`);
+      }
+    });
+  },
+  destroyed() {
+    window.removeEventListener("resize", this.onResize);
+  },
+  methods: {
+    absoluteUrl(url) {
+      if (!url) return;
+      return url.startsWith("http") ? url : new URL(this.head.url).origin + url;
+    },
+    metaValue(metaName) {
+      return findMetaContent(this.head, metaName);
+    },
+    propertyValue(propName) {
+      return findMetaProperty(this.head, propName);
+    },
+    getPreviewUrl({ imageDimensions }) {
+      const params = new URLSearchParams();
+      params.set("title", this.og.title);
+      params.set("url", getHostName(this.head.url));
+      params.set("image", this.og.image);
```

Comment:
use same conditional trick here as in Whatsapp so you don't need the `imageIsBig`